### PR TITLE
Remove state from state machine dict - UnixRemote conditionally deletes PROXY_PC

### PR DESCRIPTION
**Pull request by marcin-usielski**

### CODE REVIEW

File: moler/helpers.py [hunk 1]
```diff
@@ -574,3 +574,92 @@ def regexp_without_anchors(regexp):
...
+    for from_state in source_sm.keys():
+        for to_state in source_sm[from_state].keys():
+            if to_state == state_to_remove:
+                from_states.append(from_state)
+
+    for to_state in source_sm[state_to_remove].keys():
+        if to_state == state_to_remove:
+            continue
+        for new_from in from_states:
+            if new_from != to_state and new_from != state_to_remove:
```
Comment (tkrol88) [inline moler/helpers.py:600]: is it necessary? ` new_from != state_to_remove` I mean, is it possible situation that  ` new_from == state_to_remove`?

File: moler/device/unixremote3.py [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+    ::
+
+
+            Example of device in yaml configuration file:
+            - with PROXY_PC:
+            UNIX_1:
+            DEVICE_CLASS: moler.device.UnixRemote3.UnixRemote3
+            CONNECTION_HOPS:
+                PROXY_PC:
```
Comment (tkrol88) [inline moler/device/unixremote3.py:34]: please fix indent formatting

File: moler/helpers.py [hunk 3]
```diff
@@ -574,3 +574,92 @@ def regexp_without_anchors(regexp):
...
+    :param state_to_remove: name of state to remove
+    :return: tuple with 2 dicts without state_to_remove, 0 - new state machine, 1 - new transitions
+    """
+    from pprint import pformat
+    print(f"source sm: {pformat(source_sm)}")
+    print(f"source transitions: {pformat(source_transitions)}")
+    new_sm = copy.deepcopy(source_sm)
+    new_transitions = copy.deepcopy(source_transitions)
+
+    from_states = []
```
[the remark this answers is not in the document]
Comment (marcin-usielski) [reply moler/helpers.py]: Changed: `states_from_state_to_remove = []`

File: moler/helpers.py (hunk 1 above)
Comment (marcin-usielski) [reply moler/helpers.py]: You're right 👍

File: moler/helpers.py [hunk 4]
```diff
@@ -574,3 +574,89 @@ def regexp_without_anchors(regexp):
...
+                ],
+            }
+
+    del_state_to_remove(sm=new_sm, state_to_remove=state_to_remove)
+    del_state_to_remove(sm=new_transitions, state_to_remove=state_to_remove)
+
+    return (new_sm, new_transitions)
+
+
+def del_state_to_remove(sm: dict, state_to_remove: str) -> None:
```
Comment (Ernold11) [inline moler/helpers.py:622]: maybe just delete state?

Comment (marcin-usielski) [reply moler/helpers.py:622]: `_delete_state`

File: moler/helpers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcin-usielski) [reply moler/helpers.py]: done

File: moler/helpers.py (hunk 1 above)
Comment (marcin-usielski) [reply moler/helpers.py]: done

File: moler/device/unixremote3.py (hunk 2 above)
Comment (marcin-usielski) [reply moler/device/unixremote3.py]: done

**Pull request merged**